community/kernel/src/main/java/org/neo4j/kernel/impl/api/index/BatchingMultipleIndexPopulator.java

```diff
@@ -131,14 +129,11 @@ protected IndexPopulation createPopulation( IndexPopulator populator,
     @Override
     protected void populateFromQueue( long currentlyIndexedNodeId )
     {
-        if ( queue.size() >= QUEUE_THRESHOLD )
-        {
-            log.debug( "Populating from queue." + EOL + this );
-            flushAll();
-            awaitCompletion();
-            super.populateFromQueue( currentlyIndexedNodeId );
-            log.debug( "Drained queue and all batched updates." + EOL + this );
-        }
+        log.debug( "Populating from queue." + EOL + this );
+        flushAll();
```

Comment:
@tinwelint no no, now flushing condition moved up to MultipleIndexPopulator and we will call `populateFromQueue` now only when `isQueueThresholdReached` will allow us to do that; so in terms of behaviour it should be exactly the same